src/modules/commander/Commander.cpp

```diff
@@ -1950,7 +2078,15 @@ Commander::run()
 		}
 
 		// update manual_control_setpoint before geofence (which might check sticks or switches)
-		_manual_control_setpoint_sub.update(&_manual_control_setpoint);
+		if (_manual_control_setpoint_sub.update(&_manual_control_setpoint)) {
+			// manual control from mavlink
+			if (_manual_control_setpoint.data_source > manual_control_setpoint_s::SOURCE_RC) {
+				// if there's never been a mode change force position control as initial state
+				if (!_armed.armed && (_internal_state.main_state_changes == 0)) {
+					_internal_state.main_state = commander_state_s::MAIN_STATE_POSCTL;
```

Comment:
@dagar / @MaEtUgR only thing i am not 100% sure is if it's better to also add here a `_status_changed = true` (that would trigger a blink in the led), to make it consistent with the logic used for `main_state_transition`()

Reply:
So this part is relying on the `main_state_changes` counter increasing every time the main state is changed while overriding the `main_state_transition()` function (which would increase the value) and setting the main state without any checks. This will likely break with the next related change 😬

Reply:
The counter is a good idea but overriding `main_state_transition()` is a workaround...

Reply:
As discussed it's better than adding a different state to cover it and leads the discussion into converting to a clear user intended state/mode and available fallback.